components/infra-proxy-service/server/v1/orgs_test.go

```diff
@@ -0,0 +1,908 @@
+			})
+			require.NoError(t, err2)
+			require.NotNil(t, resp)
+			assert.Equal(t, resp1.Org.Id, resp.Org.Id)
+			assert.Equal(t, resp1.Org.Name, resp.Org.Name)
+
+			orgsAfter, err := cl.GetOrgs(context.Background(), &request.GetOrgs{
+				ServerId: serverRes.Server.Id,
+			})
+			require.NoError(t, err)
+			assert.Equal(t, len(orgsAfter.Orgs), len(orgsBefore.Orgs)-1)
+
+			authzMock.PurgeSubjectFromPoliciesFunc = test.DefaultMockPurgeFunc
+			cleanupOrg(ctx, t, cl, resp2.Org.Id)
+		})
+
+		t.Run("when an existing org is filtered by projects return NotFound", func(t *testing.T) {
+			authzMock.PurgeSubjectFromPoliciesFunc = func(
+				_ context.Context, req *authz.PurgeSubjectFromPoliciesReq) (*authz.PurgeSubjectFromPoliciesResp, error) {
+				if req.Subject == "team:local:*" {
+					return &authz.PurgeSubjectFromPoliciesResp{}, nil
+				}
+				return nil, errors.New("unexpected org name passed to PurgeSubjectFromPolicies")
+			}
+
+			secretsMock.EXPECT().Create(gomock.Any(), &newSecret, gomock.Any()).Return(secretID, nil)
+			secretsMock.EXPECT().Read(gomock.Any(), secretID, gomock.Any()).Return(&secretWithID, nil)
+			secretsMock.EXPECT().Delete(gomock.Any(), secretID, gomock.Any())
+
+			req := &request.CreateOrg{
+				Name:      "4thcoffee",
+				AdminUser: "admin",
+				AdminKey:  "--KEY--",
+				ServerId:  serverRes.Server.Id,
+				Projects:  []string{"project1", "project2"},
+			}
+			resp1, err := cl.CreateOrg(ctx, req)
+			require.NoError(t, err)
+			require.NotNil(t, resp1)
+
+			secretsMock.EXPECT().Create(gomock.Any(), &newSecret, gomock.Any()).Return(secretID, nil)
+			secretsMock.EXPECT().Read(gomock.Any(), secretID, gomock.Any()).Return(&secretWithID, nil)
+			secretsMock.EXPECT().Delete(gomock.Any(), secretID, gomock.Any())
+			req = &request.CreateOrg{
+				Name:      "other-org",
+				AdminUser: "admin",
+				AdminKey:  "--KEY--",
+				ServerId:  serverRes.Server.Id,
+				Projects:  []string{"project2"},
+			}
+			resp2, err := cl.CreateOrg(ctx, req)
+			require.NoError(t, err)
+			require.NotNil(t, resp2)
+
+			orgsBefore, err := cl.GetOrgs(ctx, &request.GetOrgs{
+				ServerId: serverRes.Server.Id,
+			})
+			require.NoError(t, err)
+			assert.Equal(t, 2, len(orgsBefore.Orgs))
+
+			ctx = test.InsertProjectsIntoNewContext([]string{"other_project"})
+
+			resp, err2 := cl.DeleteOrg(ctx, &request.DeleteOrg{
+				Id:       resp1.Org.Id,
+				ServerId: serverRes.Server.Id,
+			})
+			require.Nil(t, resp)
+			grpctest.AssertCode(t, codes.NotFound, err2)
+
+			cleanupOrg(ctx, t, cl, resp1.Org.Id)
+			cleanupOrg(ctx, t, cl, resp2.Org.Id)
+		})
+
+		t.Run("when the org to delete is not found", func(t *testing.T) {
+			ctx := context.Background()
+			resp, err2 := cl.DeleteOrg(ctx, &request.DeleteOrg{
+				Id:       "97e01ea1-976e-4626-88c8-43345c5d934f",
+				ServerId: serverRes.Server.Id,
+			})
+
+			require.Nil(t, resp)
+			grpctest.AssertCode(t, codes.NotFound, err2)
+		})
+	})
+
+	t.Run("UpdateOrg", func(t *testing.T) {
```

Comment:
Similar comment to CreateOrg: consider adding tests where an error occurs if required fields are missing or server id is invalid.